app/bundles/EmailBundle/Swiftmailer/Transport/AmazonApiTransport.php

```diff
@@ -0,0 +1,861 @@
+        }
+
+        foreach ($metadata as $recipient => $mailData) {
+            $this->message->setBody($emailBody);
+            $msg        = $this->messageToArray($mauticTokens, $mailData['tokens'], true);
+            $rawMessage = $this->buildRawMessage($msg, $recipient);
+            $payload    = [
+                'Source'       => $msg['from']['email'],
+                'Destinations' => [$recipient],
+                'RawMessage'   => [
+                    'Data' => $rawMessage,
+                ],
+            ];
+
+            yield $payload;
+        }
+    }
+
+    /**
+     * @param type $msg
+     * @param type $recipient
+     *
+     * @return string
+     */
+    public function buildRawMessage($msg, $recipient)
+    {
+        $separator           = md5(time());
+        $separator_multipart = md5($msg['subject'].time());
+        $message             = "MIME-Version: 1.0\n";
+        $message .= 'Subject: '.$msg['subject']."\n";
+        $message .= 'From: '.$msg['from']['name'].' <'.$msg['from']['email'].">\n";
+        $message .= "To: $recipient\n";
+        if (count($msg['recipients']['cc']) > 0) {
+            $message .= 'Cc: '.implode(',', array_keys($msg['recipients']['cc']))."\n";
+        }
+        if (count($msg['recipients']['bcc']) > 0) {
+            $message .= 'Bcc: '.implode(',', array_keys($msg['recipients']['bcc']))."\n";
+        }
+        $message .= "Content-Type: multipart/mixed; boundary=\"$separator_multipart\"\n";
+        if (isset($msg['headers'])) {
+            foreach ($msg['headers'] as $key => $value) {
+                $message .= "$key: ".$value."\n";
+            }
+        }
+        $message .= "\n--$separator_multipart\n";
+
+        $message .= "Content-Type: multipart/alternative; boundary=\"$separator\"\n";
+        if (isset($msg['text']) && strlen($msg['text']) > 0) {
+            $message .= "\n--$separator\n";
+            $message .= "Content-Type: text/plain; charset=\"UTF-8\"\n";
+            $message .= "Content-Transfer-Encoding: base64\n";
+            $message .= "\n".wordwrap(base64_encode($msg['text']), 78, "\n", true)."\n";
+        }
+        $message .= "\n--$separator\n";
+        $message .= "Content-Type: text/html; charset=\"UTF-8\"\n";
+        $message .= "\n".$msg['html']."\n";
+        $message .= "\n--$separator--\n";
+
+        foreach ($msg['attachments'] as $attachment) {
+            $message .= "--$separator_multipart\n";
+            $message .= 'Content-Type: '.$attachment['type'].'; name="'.$attachment['name']."\"\n";
+            $message .= 'Content-Disposition: attachment; filename="'.$attachment['name']."\"\n";
+            $message .= "Content-Transfer-Encoding: base64\n";
+            $message .= "\n".$attachment['content']."\n";
+        }
+
+        return $message."--$separator_multipart--";
+    }
+
+    /**
+     * @return int
+     */
+    public function getMaxBatchLimit()
+    {
+        return 50;
```

Comment:
I'm not sure why, but using #6977 I have to set it as 49 to not exceed AWS limit.. Probably it is documented somewhere in AWS docs about what exactly counts as an recipient.

Reply:
I did not have to change this in #6977

Reply:
@heroic the original value was 10 on #6977, how was your sending speed?

Reply:
@Wise-Gorilla The send rate seems okay, coz our SES send limits are high(500/s, and 1M per day). But our emails aren't really time sensitive so doesn't matter as much to us.

Reply:
Actually what i meant is when that value was set to 10 on #6977 then how many emails per second you was able to send approximately?

Reply:
I think the limit of 50 is set because in SES, you can send to 50 recipients for each batch sent, mentioned here https://docs.aws.amazon.com/ses/latest/DeveloperGuide/quotas.html check the part that says `Maximum number of recipients per message`

Reply:
@mabumusa1 @Wise-Gorilla again, please resolve the conversation if it's dealt with if not, a status update would be helpful.

Reply:
I'm leaving the limit at 50 as per the official Amazon docs. Since Amazon counts anyone in the To, CC and BCC fields against this limit, it might cause issues if Mautic e.g. looks at only the "To" field. But let's merge this for now. If anyone runs into problems with this setting, please feel free to submit a bug report. Mautic will throw an exception if this happens so it's easy to troubleshoot then.